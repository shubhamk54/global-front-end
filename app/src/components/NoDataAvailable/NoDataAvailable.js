import PropTypes from 'prop-types';
import React from 'react';
import './NoDataAvailable.scss';

function NoDataAvailable({ ...props }) {
    const { title, subTitle } = props;

    return <section className='no-data'>
        <p className='no-data-title'>{title}</p>
        <p className='no-data-subTitle'>{subTitle}</p>
    </section>;
}

NoDataAvailable.defaultProps = {
    title: 'Data not available',
    subTitle: '',
};

NoDataAvailable.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
};


export default NoDataAvailable;