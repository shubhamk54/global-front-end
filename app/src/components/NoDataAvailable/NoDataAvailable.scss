.no-data {
  text-align: center;
  padding: 50px;

  &-title {
    font-size: 21px;
    font-weight: 520;
  }
  &-subTitle {
    font-size: 15px;
    font-weight: 300;
  }
}
