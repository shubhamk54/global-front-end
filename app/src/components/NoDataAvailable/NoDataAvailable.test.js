import React from 'react';
import NoDataAvailable from './NoDataAvailable.js';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

const fakeProps = {
    title: 'Data not available',
    subTitle: '',
};

describe('Snapshot testing for NoDataAvailable component', () => {

    it('matches the snapshot', () => {
        const tree = renderer.create(<NoDataAvailable {...fakeProps} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

test('NoDataAvailable renders the title inside it', () => {
    const wrapper = mount(
        <NoDataAvailable {...fakeProps} />
    );
    const title = wrapper.find('.no-data-title');
    expect(title.text()).toEqual(fakeProps.title);
});

test('NoDataAvailable renders the sub-title inside it', () => {
    const wrapper = mount(
        <NoDataAvailable {...fakeProps} />
    );
    const subTitle = wrapper.find('.no-data-subTitle');
    expect(subTitle.text()).toEqual(fakeProps.subTitle);
});

test('NoDataAvailable renders section, title, and sub-title', () => {
    const wrapper = mount(
        <NoDataAvailable {...fakeProps} />
    );
    const section = wrapper.find('.no-data');
    const title = wrapper.find('.no-data-title');
    const subTitle = wrapper.find('.no-data-subTitle');

    expect(section.text()).toBeTruthy();
    expect(title.text()).toEqual(fakeProps.title);
    expect(subTitle.text()).toEqual(fakeProps.subTitle);
});
