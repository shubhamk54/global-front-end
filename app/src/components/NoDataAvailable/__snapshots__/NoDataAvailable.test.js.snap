// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for NoDataAvailable component matches the snapshot 1`] = `
<section
  className="no-data"
>
  <p
    className="no-data-title"
  >
    Data not available
  </p>
  <p
    className="no-data-subTitle"
  >
    
  </p>
</section>
`;
